### PR TITLE
Issue 180: Gamma delay convergence and parameter recovery

### PR DESCRIPTION
**Pull request by athowes**
## Description

This draft PR will close issue #180. It will be finished once PR #183 is done.

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by seabbs** [comment]
I'm seeing warnings due to lack of gamma priors and a failure in the new predict params fn. Generally the tests are overly noise as Stan progress needs silencing. Latter two are likely new issues

File: tests/testthat/test-int-latent_individual.R [hunk 1]
```diff
@@ -85,6 +85,43 @@ test_that("epidist.epidist_latent_individual Stan code has no syntax errors and
   expect_no_error(mod_gamma$compile())
 })
 
+test_that("epidist.epidist_latent_individual fits and the MCMC converges in the gamma delay case", { # nolint: line_length_linter.
+  # Note: this test is stochastic. See note at the top of this script
+  skip_on_cran()
+  set.seed(1)
+  fit_gamma <- epidist(
```
Comment (seabbs) [inline tests/testthat/test-int-latent_individual.R:92]: We aren't passing a seed to brms?

File: tests/testthat/test-int-latent_individual.R [hunk 2]
```diff
@@ -85,6 +85,43 @@ test_that("epidist.epidist_latent_individual Stan code has no syntax errors and
...
+    data = prep_obs_gamma,
+    family = stats::Gamma(link = "log"),
```
Comment (seabbs) [inline tests/testthat/test-int-latent_individual.R:94]: Here and elsewhere I think we should be silencing progress messages during testing
 Not doing so makes the output hard to pass as it is so noisy.

**Comment by athowes** [comment]
> I'm seeing warnings due to lack of gamma priors and a failure in the new predict params fn. Generally the tests are overly noise as Stan progress needs silencing. Latter two are likely new issues

The `newdata` prediction failure is:

```

**Comment by athowes** [comment]
(Oops, that closing the PR was a misclick)

**Review by seabbs — APPROVED** [review]
This looks good. Nice spot. To make this easier to review/trust a bit more context in the review (i.e a comment as to the final fix) + a plot would have been really nice.

**Comment by seabbs** [comment]
I assume the test fail is stochastic and so the test setup being used needs hardening.

File: tests/testthat/test-int-latent_individual.R (hunk 2 above)
Comment (athowes) [reply tests/testthat/test-int-latent_individual.R:94]: Do you have suggestions about how to do that? I'm looking in `epinowcast` now

File: tests/testthat/test-int-latent_individual.R (hunk 1 above)
Comment (athowes) [reply tests/testthat/test-int-latent_individual.R:92]: Good spot, if it's not there then it should be and I'll add it

**Comment by athowes** [comment]
> I assume the test fail is stochastic and so the test setup being used needs hardening.

What does it mean to harden the test set-up? What do I do to do that?

**Comment by seabbs** [comment]
> What does it mean to harden the test set-up? What do I do to do that?

Make it less susceptible to stochastic variation. setting seeds more draws etc

**Review by seabbs — APPROVED** [review]

**Pull request merged**